cpp/memilio/epidemiology/state_age_function.h

```diff
@@ -456,6 +693,104 @@ struct ConstantFunction : public StateAgeFunction {
     }
 };
 
+/**
+ * @brief Class that defines an Erlang density function with the parameters shape and scale depending on the state age.
+ * Class is needed for the initialization of the subcompartments for LCT model.
+ * ErlangDensity is derived from StateAgeFunction. 
+ * The shape parameter of the Erlang function is the parameter of the Stateagefunction. 
+ * Attention: The density does not have the characteristics of a TransitionDistribution!!
+ * The Function is of the StateAgeFunction-Type b).
+ */
+struct ErlangDensity : public StateAgeFunction {
+
+    /**
+     * @brief Constructs a new ErlangDensity object.
+     * 
+     * @param[in] init_shape Parameter shape of the ErlangDensity. For the Erlang distribution, shape has to be a positive integer.
+      * @param[in] init_scale Parameter scale of the ErlangDensity. Corresponds to the inverse rate parameter.
+     */
+    ErlangDensity(unsigned int init_shape, ScalarType init_scale)
+        : StateAgeFunction(init_shape, 0., init_scale)
+    {
+    }
+
+    /**
+     * @brief Defines ErlangDensity depending on state_age.
+     *
+     * Parameters scale and shape are used.
+     * 
+     * @param[in] state_age Time at which the function is evaluated.
+     * @return Evaluation of the function at state_age. 
+     */
+    ScalarType eval(ScalarType state_age) override
+    {
+        if (state_age < 0) {
+            return 0;
+        }
+        int shape = (int)m_parameter;
+        return std::pow(state_age / m_scale, shape - 1) / (m_scale * boost::math::factorial<ScalarType>(shape - 1)) *
+               std::exp(-state_age / m_scale);
+    }
+
+    /**
+     * @brief Computes the maximum of the support of the function. 
+     * 
+     * For small time steps and small variance of the density it is possible that dt is returned with the function of StateAgeFunction.
+     * StateAgeFunction is designed for survialfunctions, not for densities.
+     * Therefore with this function we calculate the smallest time value t where function(tau)=0 for all tau>t.
+     *
+     * @param[in] dt Time step size. 
+     * @param[in] tol Tolerance used for cutting the support if the function value falls below. 
+     * @return ScalarType support_max
+     */
+    ScalarType get_support_max(ScalarType dt, ScalarType tol = 1e-10) override
+    { // We are looking for the smallest time value t where function(tau)=0 for all tau>t. Thus support max is bigger
+        //  than the mean.
```

Comment:
Are you using here that the density is monotonically decreasing for tau > mean?

Reply:
Yes, added this.